engines/wasmtime/build.rs

```diff
@@ -0,0 +1,189 @@
+    // (and thus faster) than cloning the entire repository.
+    section("Retrieving the repository");
+    exec(&["git", "init"], &build_dir);
+    exec(&["git", "remote", "add", "origin", &repository], &build_dir);
+    exec(
+        &["git", "fetch", "--depth", "1", "origin", &revision],
+        &build_dir,
+    );
+    exec(&["git", "checkout", "FETCH_HEAD"], &build_dir);
+    exec(
+        &["git", "submodule", "update", "--init", "--depth", "1"],
+        &build_dir,
+    );
+
+    // Build the engine library.
+    section("Building the engine");
+    exec(
+        &["cargo", "build", "--release", "-p", "wasmtime-bench-api"],
+        &build_dir,
+    );
+
+    // Construct a `.build-info` file that will capture the important details a user would want to
+    // know if attempting to replicate benchmark results. (The current set is not exhaustive!).
+    section("Collecting metadata");
+    let build_info = write_buildinfo(&build_dir, &repository, &revision);
+    let build_info_contents =
+        fs::read_to_string(&build_info).expect("unable to read .build-info file");
+    eprintln!("{}", build_info_contents);
+
+    // Finally, the generated files are copied to their destination and we clean up the build
+    // directory.
+    section("Copying files to destination");
+    let from_engine_library = build_dir
+        .join("target/release")
+        .join(as_library_filename("wasmtime_bench_api"));
+    let to_engine_library = destination_dir.join(as_library_filename("engine"));
+    copy(from_engine_library, to_engine_library);
+    let to_build_info = destination_dir.join(".build-info");
+    copy(build_info, to_build_info);
+    if remove_build_dir {
+        eprintln!(
+            "Removing temporary build directory: {}",
+            build_dir.display()
+        );
+        fs::remove_dir_all(&build_dir).expect("unable to clean up temporary build directory");
+    }
+}
+
+/// Print a section header for logging.
+fn section(title: &str) {
+    eprintln!();
+    eprintln!("===== {} =====", title);
+}
+
+/// Helpful wrapper to access an environment variable as a string. `env::var` returns an error when
+/// the `OsString` cannot be converted, which is not exactly what we want. This function panics if the
+/// string cannot be converted but still returns an `Option` indicating if the variable was present.
+fn var(key: &str) -> Option<String> {
+    env::var_os(key).map(|s| {
+        s.into_string()
+            .expect("the given value could not be converted to UTF-8")
+    })
+}
+
+/// Helpful wrapper to create a temporary directory; e.g., `/tmp/sightglass-wasmtime-build-<current
+/// unix seconds>`)
+fn create_temp_directory() -> PathBuf {
+    let mut p = env::temp_dir();
+    let time = SystemTime::now()
+        .duration_since(SystemTime::UNIX_EPOCH)
+        .unwrap();
+    p.push(format!("sightglass-wasmtime-build-{}", time.as_secs()));
```

Comment:
Ideally we'd use the `tempfile` crate here but I understand why it's desirable to have no deps for a single script file; in that case, can we at least use seconds and nanoseconds? I can imagine otherwise a situation arising where parallel builds started at once (in the same second) could conflict.